File: LeetCode/Easy/0225-implement-stack-using-queues/0225-implement-stack-using-queues.java
import java.util.LinkedList;
import java.util.Queue;

class MyStack {

        private Queue<Integer> mainQueue;
        private Queue<Integer> supportQueue;


        public MyStack() {
            mainQueue = new LinkedList<>();
            supportQueue = new LinkedList<>();
        }

        public void push(int x) {
            mainQueue.add(x);

            while(!supportQueue.isEmpty()) {
                mainQueue.add(supportQueue.poll());
            }

            Queue<Integer> temp = mainQueue;
            mainQueue = supportQueue;
            supportQueue = temp;
        }

        public int pop() {
            return supportQueue.poll();
        }

        public int top() {
            return  supportQueue.peek();
        }

        public boolean empty() {
            return supportQueue.isEmpty();
        }
    }

File: LeetCode/Easy/0225-implement-stack-using-queues/0225-implement-stack-using-queues.py
from queue import Queue

class MyStack:

    def __init__(self):
        self.main_queue = Queue()
        self.support_queue = Queue()

    def push(self, x: int) -> None:
        self.main_queue.put(x)

        while not self.support_queue.empty():
            self.main_queue.put(self.support_queue.get())
        
        self.main_queue, self.support_queue = self.support_queue, self.main_queue

    def pop(self) -> int:
        return self.support_queue.get()

    def top(self) -> int:
        return self.support_queue.queue[0]

    def empty(self) -> bool:
        return self.support_queue.empty()